### PR TITLE
feat(props): preserve comments in updated gradle.properties files

### PR DESCRIPTION
**Pull request by cfieber**

### CODE REVIEW

**Review by robzienert — APPROVED** [review]

File: src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    fun saveResult(lines: List<String>) {
+        FileWriter(propertiesFile.toFile(), Charset.forName("8859_1")).use { fw ->
+            lines.forEach { line ->
+                fw.write(line)
+                fw.write("\n")
+            }
+        }
+    }
```
Comment (plumpy) [inline src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt]:
```suggestion
        Files.write(propertiesFile, lines, Charsets.ISO_8859_1)
```

File: src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.io.FileWriter
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.regex.Pattern
+
+class NonDestructivePropertiesEditor(internal val propertiesFile: Path) {
+    internal val propsCharset: Charset = Charset.forName("8859_1")
```
Comment (plumpy) [inline src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt]: use `kotlin.text.Charsets.ISO_8859_1`

File: src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package io.spinnaker.bumpdeps
+
+import java.io.FileWriter
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.regex.Pattern
+
+class NonDestructivePropertiesEditor(internal val propertiesFile: Path) {
```
Comment (plumpy) [inline src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt:9]: I'd just inline `saveResult` into `Main.kt` and then make `propertiesFile` be a parameter of the first `updateProperty` method... I don't think having it as a property of this class really adds anything...

**Review by plumpy — APPROVED** [review]

File: src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt (hunk 3 above)
Comment (cfieber) [reply src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt:9]: The other thing it is uses for is the `updateProperties(key, value)` (`updateProperty(lines, key, value)` was mostly for testability more than anything)

I'll fix the saveResult bit but keep the ownership of file io inside this class unless you feel strongly the other direction

Comment (plumpy) [reply src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt:9]: k, fair enough, thanks!

**Pull request merged**